holidays/calendars/islamic.py

```
@@ -4264,3 +4267,83 @@ def tasua_dates(self, year: int) -> Iterable[tuple[date, bool]]:
+    
+    Criteria calculated at sunset on the 29th day of the lunar month:
+    - Moon altitude >= 3.0 degrees
+    - Elongation (Moon-Sun separation) >= 6.4 degrees
+    
+    Primary reference tracking uses Singapore's calculation baseline.
+    """
+
+    LATITUDE = 1.3521
+    LONGITUDE = 103.8198
+
+    def _get_solar_lunar_positions(self, days_since_j2000: float):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_get_solar_lunar_positions`

(ANN202)